### PR TITLE
added example for FIT

### PR DESCRIPTION
**Pull request by aopy**

### CODE REVIEW

**Comment by EtienneCmb** [comment]
Hi @aopy. You pushed to `conn_fit` to xfrites, therefore the code can not import the function. Prior to the example, we should first include the `conn_fit` function, then the unit tests and finally the example.

Can you append to this PR the code you made for the PR inside xfrites?

**Comment by EtienneCmb** [comment]
Thx @aopy. Can you check [code quality](https://github.com/brainets/frites/actions/runs/5346194814/jobs/9692857434?pr=59#step:7:1)? You need to install [flake8](https://pypi.org/project/flake8/) and, inside a terminal, go to frites directly and run `flake8` to fix the issues related to code quality.

**Comment by EtienneCmb** [comment]
Also, net computations is not correct for intracranial EEG data. In `conn_fit`, can you replace the block:

```python
  if net:
      # get unique pairs
      i_s, i_t = np.stack((i_s, i_t))[:, i_s < i_t]
      # computes net transfer
      fit = fit[i_s, :] - fit[i_t, :]
      roi_p = np.array(roi_p)[i_s]
```

by : 

```python
  if net:
      roi_po = roi_p.copy()
      done, roi_p, i_st = [], [], []
      for n_s, (s, t) in enumerate(zip(i_s, i_t)):
          # ignore if indices have already been stored
          if ([s, t] in done) or ([t, s] in done): continue  # noqa
          # find [source, target] and [target, source]
          result = np.where((i_s == t) & (i_t == s))[0]
          assert len(result) == 1
          n_t = result[0]
          # decide if (x->y - y->x) or (y->x - x->y)
          r_s, r_t = roi_po[n_s], roi_po[n_t]
          if not np.all(np.array([r_s, r_t]) == np.sort([r_s, r_t])):
              n_s, n_t = n_t, n_s
          r_s, r_t = roi_po[n_s].split('->')[0], roi_po[n_t].split('->')[0]
          # store results
          i_st.append([n_s, n_t])
          roi_p.append(f"{r_s}-{r_t}")
          done.append([s, t])
      # computes net transfer
      i_s, i_t = np.array(i_st).T
      fit = fit[i_s, :] - fit[i_t, :]
```
Thank you in advance,

File: frites/conn/tests/test_conn.py [hunk 1]
```diff
@@ -155,6 +155,22 @@ def test_conn_pid(self):
...
+        ss = StimSpecAR()
+        x = ss.fit(ar_type=ar_type, n_epochs=n_epochs, n_stim=n_stim)
+
+        for m in ['cc', 'cd']:  # mi_type
+            for n in [True, False]:  # net
+                for ad in [True, False]:  # avg_delay
+                    fit = conn_fit(x, y='trials', roi='roi', times='times',
+                                   mi_type=m, max_delay=.3, net=n,
+                                   verbose=False, avg_delay=ad)
+                    print(fit.shape)
```
Comment (EtienneCmb) [inline frites/conn/tests/test_conn.py:172]: Here, can you put an assert to test that the shape of the fit is actually changing? Something like:

```python
if n:
    assert len(fit['roi']) == 1
else:
    assert len(fit['roi']) == 2

if ad:
    assert fit.ndim == 3
else:
    assert fit.ndim == 4
```

File: frites/conn/conn_fit.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        i_s, i_t = np.array(i_st).T
+        fit = fit[i_s, :] - fit[i_t, :]
+
+    # xarray conversion
+    if avg_delay:
+        dims, coords = ('roi', 'delays', 'times'), (roi_p, delay, times)
+    else:
+        dims, coords = ('roi', 'times'), (roi_p, times)
+    attrs['max_delay'] = max_delay
+    attrs['mi_type'] = mi_type
```
Comment (EtienneCmb) [inline frites/conn/conn_fit.py:223]: Here can you add (those are the attributes of the `conn_fit` function. When saving the results, those attributes are conserved which is nice to share self-contained files with the configuration used):

```python
attrs['net'] = net
attrs['avg_delay'] = avg_delay
attrs['unit'] = 'Bits'
```

File: examples/conn/plot_fit.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+"""
+FIT: Feature specific information transfer
+==================================================================
+
+This example illustrates how to discover a feature (e.g. sensory stimuli) in information transfer between brain regions.
```
Comment (EtienneCmb) [inline examples/conn/plot_fit.py:5]: This line is probably too long and might not render correctly in the gallery of examples. Can you make it 79 characters long? Same for the `==================================================================`, you can stop it under the final letter "r" of *transfer*

**Review by EtienneCmb — CHANGES_REQUESTED** [review]
Also, we need to put the `conn_fit` in the [API tab of the doc](https://brainets.github.io/frites/api/api_connectivity.html#module-frites.conn). Below [conn_pid](https://github.com/brainets/frites/blob/master/docs/source/api/api_connectivity.rst?plain=1#L24), can you add `conn_fit` please?

**Review by EtienneCmb — CHANGES_REQUESTED** [review]
Finally, can you add your contribution to [What's new tab](https://brainets.github.io/frites/overview/ovw_whatsnew.html)? Add a line below in the [whatsnew file](https://github.com/brainets/frites/blob/master/docs/source/overview/ovw_whatsnew.rst?plain=1#L14), something like :

```
- New function :func:`frites.conn.conn_fit` to estimate the feature specific information transfer (:PR:`59`) - :author:`aopy`
```

File: frites/conn/tests/test_conn.py (hunk 1 above)
Comment (aopy) [reply frites/conn/tests/test_conn.py:172]: "assert fit.ndim == 4" fails (it is 2).

Comment (EtienneCmb) [reply frites/conn/tests/test_conn.py:172]: You're right, should be : 
```python
if ad:
    assert fit.ndim == 2
else:
    assert fit.ndim == 3
```

**Comment by EtienneCmb** [comment]
All lights are green @aopy !!! Thank you so much :+1:

**Comment by aopy** [comment]
@EtienneCmb no problem :) thanks for your input!

**Pull request merged**